checker_test.h

```diff
@@ -0,0 +1,20 @@
+#ifndef CHECKER_TEST_H
+#define CHECKER_TEST_H
+
+typedef struct
+{
+    float temperature;
+    float stateOfCharge;
+    float chargeRate;
+    int expectedBattStatus;
+
+} BatteryAttributes;
+
+#define IN_LIMIT    1
+#define NOT_IN_LIMIT    0
+
+#define NO_OF_TESTCASES 8

```

Comment:
Hard to remember to update this when adding a test... esp since it is in a different file. Try initializing the array without the size in the box-brackets.

Reply:
it works, when sizeof calculation resides in the same translation unit(file). not otherwise. Updated code to calculate number of tests.